core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java

```diff
@@ -1712,6 +1712,22 @@ private static String toSql(RelNode root, SqlDialect dialect) {
     sql(query).dialect(MysqlSqlDialect.DEFAULT).ok(expected);
   }
 
+  @Test public void testMySqlCastToVarcharWithLessThanMaxPrecision() {
+    final String query = "select cast(\"product_id\" as varchar(50)), \"product_id\" "
+        + "from \"product\" ";
+    final String expected = "SELECT CAST(`product_id` AS CHAR(50)), `product_id`\n"
+        + "FROM `foodmart`.`product`";
+    sql(query).dialect(mySqlDialect(NullCollation.HIGH)).ok(expected);
```

Comment:
Rather than `.dialect(mySqlDialect(NullCollation.HIGH))`, is `withMysql()` sufficient?

Reply:
@julianhyde I have changed the test to use`withMysql()`